api-tests/src/test/java/org/openmrs/module/reporting/report/definition/service/ReportDefinitionServiceImplTest.java

```diff
@@ -39,11 +40,32 @@
 	@Test
 	public void purgeDefinition_shouldPurgeReportDesigns() throws Exception {
 		executeDataSet("org/openmrs/module/reporting/include/ReportDefinitionServiceImplTest.xml");
-    	
+
     	ReportDefinition definition = reportDefinitionService.getDefinition(80);
     	reportDefinitionService.purgeDefinition(definition);
 	    
-    	ReportDesign reportDesign = reportService.getReportDesign(2);
-    	assertThat(reportDesign, nullValue());
+		assertThat(reportService.getReportDesign(3), nullValue());
+		assertThat(reportService.getReportDesign(4), nullValue());
+	}
+
+	/**
+	 * @see ReportDefinitionServiceImpl#purgeDefinition(String)
+	 * @verifies purge report definition by uuid and associated report designs and requests
+	 */
+	@Test
+	public void purgeDefinition_shouldPurgeDesignsAndRequestsAndDefinition() throws Exception {
+		executeDataSet("org/openmrs/module/reporting/include/ReportDefinitionServiceImplTest.xml");
+		assertNotNull(reportService.getReportDesign(3));
+		assertNotNull(reportService.getReportDesign(4));
+		assertNotNull(reportService.getReportRequest(1));
+		assertNotNull(reportService.getReportRequest(3));
+
+		reportDefinitionService.purgeDefinition("c11f5354-9567-4cc5-b3ef-163e28873926");
+
+		assertNull(reportService.getReportDesign(3));
+		assertNull(reportService.getReportDesign(4));
+		assertNull(reportService.getReportRequest(1));
+		assertNull(reportService.getReportRequest(3));
+		assertNull(reportDefinitionService.getDefinition(80));

```

Comment:
What is the use of the last assertion on line 69?

Reply:
It is meant to check that the report definition was deleted as expected.